python/pyarrow/types.pxi

```diff
@@ -2386,25 +2426,102 @@ def struct(fields):
     return pyarrow_wrap_data_type(struct_type)
 
 
+cdef _extract_union_params(children_fields, type_codes,
```

Comment:
This is super nit-picky, but should the first arg be `child_fields`? That would match what's in the body when iterating over them `child_fields` yields a single `child_field` (as opposed to yielding a `children_field`).